src/bootstrap/build/util.rs

```diff
@@ -16,6 +16,26 @@ use std::process::Command;
 use bootstrap::{dylib_path, dylib_path_var};
 use filetime::FileTime;
 
+#[cfg(not(windows))]
+pub fn build_path(path: &str) -> PathBuf {
+    PathBuf::from(path)
+}
+
+#[cfg(windows)]
+pub fn build_path(path: &str) -> PathBuf {
```

Comment:
That's only the Windows paths in msys form. There are also things like `/home` and even `/h/path` where `/h` as a directory and not a drive.